clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1227,13 +1227,14 @@ public void close(Duration timeout) {
 
     private void close(Duration timeout, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
+        boolean wasInterrupted = Thread.interrupted();
```

Comment:
the classic consumer does not reset the interrupted status, so this may introduce some behavior change to consumer listener? for example, consumer listener CAN'T see the interrupt anymore?